https://public.tableau.com/views/Milestone1_17152535083640/Dashboard19?:language=en-GB&publish=yes&:sid=&:display_count=n&:origin=viz_share_link
# **Milestone 1**

============================================

Nama        : Frederick Kurniawan Putra

Batch       : HCK016

Objective   : Notebook ini dibuat untuk melakukan eksplorasi data dan analisa mengenai hubungan musik dengan efek penyembuhan penyakit mental dengan tujuan agar bisa digunakan sebagai suplemen dalam proses terapi penyakit mental.

============================================

## **1.Background**

Institut kesehatan mental ABC adalah institut yang bergerak pada kesehatan mental yang menangani permasalahan kesehatan mental dengan melakukan kegiatan konsultasi psikolog maupun psikiater untuk pasien dengan tingkat menengah atau tinggi. Institut ini sedang memiliki project untuk menyiapkan rangkaian prosedur dan teknik yang digunakan untuk menangani gangguan mental khusus untuk pasien remaja atau professional muda. Salah satu teknik yang ingin mereka terapkan adalah dengan aktifitas mendengarkan musik sebagai alat terapi supplementary diluar sesi terapi dan obat obatan konvensional. Namun dari stakeholder institut ingin melihat apakah memang ada potensi penggunaan musik sebelum mereka menyiapkan kerangka terapi musik ini secara lebih lanjut. 

## **2. Problem Identification**

### **a. SMART Framework**

#### Specific

Memastikan bahwa mendengarkan musik cukup efektif untuk diterapkan oleh insitut sebagai salah satu teknik suplementer untuk membantu menangani gangguan mental. 

#### Measurable

Persentase pasien yang terbantu dengan mendengarkan musik terhadap gangguan mental yang dimiliki > 50% agar teknik bisa dianggap efektif untuk diterapkan.

#### Achiveable

Menganalisa efektifitas aktifitas mendengarkan musik dalam membantu penyembuhan gangguan mental. Mendapatkan insight mengenai musik jenis apa yang cocok untuk direkomendasikan kepada pasien.

#### Relevant

Analisa efektifitas musik ini akan memberikan gambaran apakah musik mampu menjadi suplemen dalam membantu penyembuhan gangguan mental. Sehingga ketika terapi musik ini diterapkan, maka persentase keberhasilan terapi ini sesuai dengan standar dan bisa membantu meningkatkan progres penyembuhan pasien gangguan mental.

#### Time-bound

Mengumpulkan jawaban questionnaire dalam 1 bulan, lalu melakukan pemrosesan data, dan melakukan analisa dalam 1 minggu.

### **b. Problem Statement**

Menganalisa dan memastikan bahwa mendengarkan musik cukup efektif untuk diterapkan sebagai suplementer dalam penanganan gangguan mental dengan efektifitas > 50% dalam kurun wakut 1 bulan dan 1 minggu.

### **c. Problem Breakdown 5W + 1H**

1. Berapa persen responden yang terbantu dengan mendengarkan musik untuk meringankan gangguan kesehatan mental?
2. Genre-genre apa saja yang digemari oleh responden?
3. Genre lagu apa saja yang dapat membantu meringkankan gangguan kesehatan mental?
4. Apakah genre favorit responden berpengaruh terhadap efektifitas musik dalam perkembangan kesehatan mental responden?
5. Berapa rata-rata lama waktu aktivitas mendengarkan musik yang dilakukan responden?
6. Apakah semakin tua umur akan mempengaruhi lamanya mendengarkan lagu?
7. Apakah lama waktu mendengarkan musik berpengaruh terhadap efektifitas dalam meringankan gangguan kesehatan mental?
8. Seperti Apa Persebaran data BPM musik terhadap masing masing efek dari mendengarkan musik?

## **3. Data Loading**

In [1]:
#import all necessary
import pandas as pd

#import data from csv file
df_raw = pd.read_csv('mxmh_survey_results.csv')

In [2]:
#simple data exploration
df_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 736 entries, 0 to 735
Data columns (total 33 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Timestamp                     736 non-null    object 
 1   Age                           735 non-null    float64
 2   Primary streaming service     735 non-null    object 
 3   Hours per day                 736 non-null    float64
 4   While working                 733 non-null    object 
 5   Instrumentalist               732 non-null    object 
 6   Composer                      735 non-null    object 
 7   Fav genre                     736 non-null    object 
 8   Exploratory                   736 non-null    object 
 9   Foreign languages             732 non-null    object 
 10  BPM                           629 non-null    float64
 11  Frequency [Classical]         736 non-null    object 
 12  Frequency [Country]           736 non-null    object 
 13  Frequ

**Insight**
* Terdapat 736 entries dan 33 kolom dalam dataset ini.
* Kolom `Primary streaming service` dan `Timestamp` tidak dibutuhkan dalam analisa kali ini, sehingga kolom tersebut akan didrop.
* Kolom `Age`, `While Working`, `Instrumentalist`, `Composer`, `Foreign languages`, `BPM`, dan `Music effects` memiliki null value, sehingga harus dihandle sesuai konteks isi kolom.
* Kolom `BPM` memiliki data type yang tidak sesuai, bisa diganti dengan integer

In [5]:
df_raw.head(10)

Timestamp   Age          Primary streaming service  Hours per day  \
0  8/27/2022 19:29:02  18.0                            Spotify            3.0   
1  8/27/2022 19:57:31  63.0                            Pandora            1.5   
2  8/27/2022 21:28:18  18.0                            Spotify            4.0   
3  8/27/2022 21:40:40  61.0                      YouTube Music            2.5   
4  8/27/2022 21:54:47  18.0                            Spotify            4.0   
5  8/27/2022 21:56:50  18.0                            Spotify            5.0   
6  8/27/2022 22:00:29  18.0                      YouTube Music            3.0   
7  8/27/2022 22:18:59  21.0                            Spotify            1.0   
8  8/27/2022 22:33:05  19.0                            Spotify            6.0   
9  8/27/2022 22:44:03  18.0  I do not use a streaming service.            1.0   

  While working Instrumentalist Composer         Fav genre Exploratory  \
0           Yes             Yes      Yes             Latin         Yes   
1           Yes              No       No              Rock         Yes   
2            No              No       No  Video game music          No   
3           Yes              No      Yes              Jazz         Yes   
4           Yes              No       No               R&B         Yes   
5           Yes             Yes      Yes              Jazz         Yes   
6           Yes             Yes       No  Video game music         Yes   
7           Yes              No       No             K pop         Yes   
8           Yes              No       No              Rock          No   
9           Yes              No       No               R&B         Yes   

  Foreign languages  ...  Frequency [R&B]  Frequency [Rap] Frequency [Rock]  \
0               Yes  ...        Sometimes  Very frequently            Never   
1                No  ...        Sometimes           Rarely  Very frequently   
2               Yes  ...            Never           Rarely           Rarely   
3               Yes  ...        Sometimes            Never            Never   
4                No  ...  Very frequently  Very frequently            Never   
5               Yes  ...  Very frequently  Very frequently  Very frequently   
6               Yes  ...           Rarely            Never            Never   
7               Yes  ...        Sometimes           Rarely            Never   
8                No  ...            Never            Never  Very frequently   
9               Yes  ...        Sometimes           Rarely        Sometimes   

  Frequency [Video game music] Anxiety Depression Insomnia  OCD Music effects  \
0                    Sometimes     3.0        0.0      1.0  0.0           NaN   
1                       Rarely     7.0        2.0      2.0  1.0           NaN   
2              Very frequently     7.0        7.0     10.0  2.0     No effect   
3                        Never     9.0        7.0      3.0  3.0       Improve   
4                       Rarely     7.0        2.0      5.0  9.0       Improve   
5                        Never     8.0        8.0      7.0  7.0       Improve   
6                    Sometimes     4.0        8.0      6.0  0.0       Improve   
7                       Rarely     5.0        3.0      5.0  3.0       Improve   
8                        Never     2.0        0.0      0.0  0.0       Improve   
9                    Sometimes     2.0        2.0      5.0  1.0       Improve   

     Permissions  
0  I understand.  
1  I understand.  
2  I understand.  
3  I understand.  
4  I understand.  
5  I understand.  
6  I understand.  
7  I understand.  
8  I understand.  
9  I understand.  

[10 rows x 33 columns]

In [109]:
#melakukan check value apa saja pada kolom bpm
df_raw['BPM'].sort_values().unique()

array([0.00000000e+00, 4.00000000e+00, 8.00000000e+00, 2.00000000e+01,
       4.00000000e+01, 5.20000000e+01, 5.50000000e+01, 5.60000000e+01,
       6.00000000e+01, 6.10000000e+01, 6.20000000e+01, 6.30000000e+01,
       6.60000000e+01, 6.80000000e+01, 7.00000000e+01, 7.20000000e+01,
       7.30000000e+01, 7.50000000e+01, 7.60000000e+01, 7.70000000e+01,
       7.80000000e+01, 7.90000000e+01, 8.00000000e+01, 8.10000000e+01,
       8.20000000e+01, 8.40000000e+01, 8.50000000e+01, 8.60000000e+01,
       8.70000000e+01, 8.80000000e+01, 8.90000000e+01, 9.00000000e+01,
       9.10000000e+01, 9.20000000e+01, 9.30000000e+01, 9.40000000e+01,
       9.50000000e+01, 9.60000000e+01, 9.70000000e+01, 9.80000000e+01,
       9.90000000e+01, 1.00000000e+02, 1.01000000e+02, 1.02000000e+02,
       1.03000000e+02, 1.04000000e+02, 1.05000000e+02, 1.06000000e+02,
       1.07000000e+02, 1.08000000e+02, 1.09000000e+02, 1.10000000e+02,
       1.11000000e+02, 1.12000000e+02, 1.13000000e+02, 1.14000000e+02,
      

In [9]:
#check if there is non standard null variable
df_raw['Age'].unique()

array([18., 63., 61., 21., 19., nan, 17., 16., 15., 14., 20., 26., 23.,
       37., 36., 24., 22., 42., 25., 32., 29., 41., 31., 33., 30., 43.,
       28., 34., 13., 35., 27., 53., 49., 60., 38., 44., 40., 64., 56.,
       59., 10., 12., 54., 57., 69., 72., 73., 39., 58., 67., 70., 55.,
       48., 71., 74., 51., 65., 80., 50., 68., 46., 89.])

In [149]:
df_raw['BPM'].unique()

array([1.56000000e+02, 1.19000000e+02, 1.32000000e+02, 8.40000000e+01,
       1.07000000e+02, 8.60000000e+01, 6.60000000e+01, 9.50000000e+01,
       9.40000000e+01, 1.55000000e+02,            nan, 1.25000000e+02,
       8.80000000e+01, 1.48000000e+02, 1.03000000e+02, 1.20000000e+02,
       9.90000000e+01, 1.18000000e+02, 7.90000000e+01, 1.69000000e+02,
       1.36000000e+02, 1.01000000e+02, 1.26000000e+02, 1.83000000e+02,
       1.42000000e+02, 7.50000000e+01, 8.90000000e+01, 1.40000000e+02,
       1.61000000e+02, 8.00000000e+01, 1.30000000e+02, 1.47000000e+02,
       7.70000000e+01, 1.15000000e+02, 1.57000000e+02, 9.00000000e+01,
       1.28000000e+02, 9.60000000e+01, 9.10000000e+01, 5.50000000e+01,
       8.50000000e+01, 1.60000000e+02, 1.08000000e+02, 1.70000000e+02,
       1.74000000e+02, 1.27000000e+02, 1.65000000e+02, 1.13000000e+02,
       1.67000000e+02, 1.10000000e+02, 1.50000000e+02, 9.30000000e+01,
       1.09000000e+02, 1.75000000e+02, 1.62000000e+02, 1.89000000e+02,
      

## **4. Data Cleaning**

In [23]:
df_clean = df_raw.copy()

In [24]:
#drop semua row yang ada null value di kolom age, while working, instrumentalist, composer, foreign languages, bpm, dan music effects
df_clean = df_clean.dropna(subset=['Age', 'While working', 'Instrumentalist', 'Composer', 'Foreign languages', 'BPM', 'Music effects'])

In [19]:
#melakukan drop entries untuk value kolom 'Age' yang null variable
df_clean = df_clean.dropna(subset=['Age'])

In [26]:
#melakukan check entries 
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 617 entries, 2 to 735
Data columns (total 33 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Timestamp                     617 non-null    object 
 1   Age                           617 non-null    float64
 2   Primary streaming service     616 non-null    object 
 3   Hours per day                 617 non-null    float64
 4   While working                 617 non-null    object 
 5   Instrumentalist               617 non-null    object 
 6   Composer                      617 non-null    object 
 7   Fav genre                     617 non-null    object 
 8   Exploratory                   617 non-null    object 
 9   Foreign languages             617 non-null    object 
 10  BPM                           617 non-null    float64
 11  Frequency [Classical]         617 non-null    object 
 12  Frequency [Country]           617 non-null    object 
 13  Frequency 

In [27]:
#ubah data type bpm ke int
df_clean['BPM'] = df_clean['BPM'].astype('int')

In [150]:
#cek value value yang ada di kolom bpm
df_clean['BPM'].unique()

array([      132,        84,       107,        86,        66,        95,
              94,       155,       125,        88,       148,       103,
             120,        99,       118,        79,       169,       136,
             101,       126,       183,       142,        75,        89,
             140,       161,        80,       130,       147,        77,
             115,       157,        90,       128,        96,        91,
              55,        85,       160,       108,       156,       170,
             174,       127,       165,       113,       167,       110,
             150,        93,       109,       175,       162,       189,
             121,       200,       105,       193,       149,       134,
             100,        98,        78,       117,       122,       179,
             166,       144,        61,       185,        92,       194,
              60,       190,        87,       171,       119,       141,
             146,       164,       145,       112, 

Terdapat value 999999999 di kolom BPM yang kemungkinan adalah input error karena secara konteks tidak masuk akal untuk responden mendengarkan lagu dengan bpm 999999999. Row yang memiliki value ini akan didrop.

In [155]:
df_clean[df_clean['BPM']>90000]

Timestamp   Age Primary streaming service  Hours per day  \
568  9/4/2022 15:41:59  16.0                   Spotify            7.0   

    While working Instrumentalist Composer         Fav genre Exploratory  \
568           Yes              No       No  Video game music         Yes   

    Foreign languages  ...  Frequency [R&B]  Frequency [Rap] Frequency [Rock]  \
568                No  ...            Never  Very frequently           Rarely   

    Frequency [Video game music] Anxiety Depression Insomnia  OCD  \
568                    Sometimes     4.0        8.0      8.0  4.0   

    Music effects    Permissions  
568     No effect  I understand.  

[1 rows x 33 columns]

Hanya ada 1 row yang akan didrop

In [156]:
#drop row dengan value kolom hasil input error
df_clean = df_clean.drop(df_clean[df_clean['BPM']> 90000.].index)

In [157]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 616 entries, 2 to 735
Data columns (total 33 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Timestamp                     616 non-null    object 
 1   Age                           616 non-null    float64
 2   Primary streaming service     615 non-null    object 
 3   Hours per day                 616 non-null    float64
 4   While working                 616 non-null    object 
 5   Instrumentalist               616 non-null    object 
 6   Composer                      616 non-null    object 
 7   Fav genre                     616 non-null    object 
 8   Exploratory                   616 non-null    object 
 9   Foreign languages             616 non-null    object 
 10  BPM                           616 non-null    int32  
 11  Frequency [Classical]         616 non-null    object 
 12  Frequency [Country]           616 non-null    object 
 13  Frequency 

In [233]:
#simpan dataframe yang sudah di cleaning ke dalam csv untuk dashboard tableau
df_clean.to_csv('mxmh_survey_results_clean.csv', index=False)

## **5. Analisis dan Perhitungan**

In [29]:
import plotly.express as px #import library plotly

### **1. Berapa persen responden yang terbantu dengan mendengarkan musik untuk meringankan gangguan kesehatan mental?**

In [219]:
df_test = pd.DataFrame(data=df_clean['Music effects'] ,columns=['Music effects']) #Buat dataframe baru dengan isi hanya kolom music effects dan semua rownya
df_test2 = df_test.groupby(['Music effects']).size().reset_index(name='People') #Grouping data berdasarkan isi kolom music effects, dan melakukan perhitungan jumlah row untuk setiap music effect
fig = px.pie(df_test2, values='People', names='Music effects', title='Persen Jumlah Responden terhadap efek mendengarkan musik')
fig.show()
df_test2

Music effects  People
0       Improve     466
1     No effect     135
2        Worsen      15

**Insight**

Mengenai insight data yang diambil, dapat dilihat
* 75.6% responden menyatakan bahwa ada improvement terhadap kesehatan mental mereka setelah mendengarkan musik.
* 21.9% responden menyatakan tidak ada efek terhadap kesehatan mental mereka.
* 2,44% responden merasa bahwa kesehatan mental mereka menjadi semakin buruk.

Disini kita bisa berasumsi bahwa mendengarkan musik mungkin bisa membantu pasien yang memiliki gangguan mental dikarenakan persentase yang merasakan perkembangan yang baik lebih besar dari 50%, standar persentase yang ditetapkan. Namun hal ini perlu kita analisa lebih dalam seperti apa improvement dan musik apa yang menjadi pengaruh besar terhadap improvement tersebut.

### **2. Genre-genre apa saja yang digemari oleh responden?**

In [235]:
df_test = pd.DataFrame(data=df_clean['Fav genre'] ,columns=['Fav genre']) #Buat dataframe baru dengan kolom 'Fav genre' dan isinya
df_test2 = df_test.groupby(['Fav genre']).size().reset_index(name='People').sort_values(by='People') #Mengelompokkan value jumlah row berdasarkan 'Fav genre'
fig = px.bar(df_test2, x='People', y='Fav genre', title='Perbandingan Jumlah Responden berdasarkan Genre Musik Favorit')
fig.show()

**Insight**

Berdasarkan bar chart yang didapatkan, kita bisa melihat bahwa:
* Jumlah responden dengan genre Rock sebagai genre favorit adalah tertinggi dibanding kategori genre yang lain.
* Jumlah responden yang tertinggi kedua untuk genre favorit adalah Pop
* Jumlah responden yang tertinggi ketiga untuk genre favorit adalah Metal
* Jumlah responden dengan genre Latin sebagai genre favorit adalah terendah dibanding kategori genre yang lain.


### **3. Genre lagu apa saja yang dapat membantu meringkankan gangguan kesehatan mental?**

Untuk menjawab soal ini, akan dilihat persentase masing masing genre favorit responden yang menjawab adanya perkembangan dari kesehatan mental mereka setelah mendengar musik.

In [236]:
df_g4 = pd.DataFrame(data= df_clean[df_clean['Music effects'] == 'Improve']['Fav genre'],columns=['Fav genre']) #Dataframe baru dengan isi row dimana value kolom 'Music effects' = improve
df_g44 = df_g4.groupby(['Fav genre']).size().reset_index(name='People').sort_values(by='People') #Mengelompokkan value jumlah row berdasarkan 'Fav genre'
fig = px.bar(df_g44, x='People', y='Fav genre',title='Perbandingan Jumlah Responden dengan respon improved di Music effects berdasarkan Genre Musik Favorit')
fig.show()

**Insight**

Dari visualisasi yang didapat, kita masih bisa melihat bahwa genre rock masih menjadi genre yang paling diminati oleh responden yang merasa terbantu dengan lagu yang didengar, disusul oleh Pop, Metal, dan EDM. Untuk Genre yang berada di peringkat bawah ada Rap, Lofi, Gospel, dan Latin.

### **4. Apakah genre favorit responden berpengaruh terhadap efektifitas musik dalam perkembangan kesehatan mental responden?**

Untuk Menjawab pertanyaan ini, digunakan Chi-Squared Hypothesis Test dikarenakan 2 variable merupakan categorical.

**H0**: Tidak ada hubungan antara Musik Favorit dengan Efek dari Musik.

**H1**: Ada hubungan antara Musik Favorit dengan Efek dari Musik.

Pengetesan menggunakan Confidence level 95% dengan Critical Value 0.05(5%).

In [58]:
from scipy import stats #import library

In [160]:
#contigency table untuk mencari jumlah kategori musik efek terhadap kategori genre favorit
contigency_table = pd.crosstab(df_clean['Fav genre'],df_clean['Music effects'])
contigency_table

Music effects     Improve  No effect  Worsen
Fav genre                                   
Classical              26         11       1
Country                18          3       0
EDM                    30          6       0
Folk                   20          5       0
Gospel                  4          0       0
Hip hop                28          4       0
Jazz                   16          3       0
K pop                  17          4       0
Latin                   1          1       0
Lofi                   10          0       0
Metal                  57         20       0
Pop                    75         20       2
R&B                    22          8       0
Rap                    15          4       1
Rock                  107         35       7
Video game music       20         11       4

In [161]:
res = stats.chi2_contingency(contigency_table) #memanggil fungsi chi-squared test
print("P-value:",res.pvalue)

P-value: 0.1746539809839702


Critical value: 0.05

p-val = 0.1746 > critical value

**H0 failed to rejected**

**Insight**

Dari hasil Chi-squared test, kita dapat menarik insight bahwa tidak ada hubungan antara Genre Favorit dengan efek dari mendengarkan musik. Hal ini bisa mengindikasikan bahwa genre tertentu tidak menjadi acuan untuk efektifitas untuk membantu penyembuhan gangguan mental.

### **5. Berapa rata-rata lama waktu aktivitas mendengarkan musik yang dilakukan responden?**

In [162]:
print(f"Mean dari data berapa jam per hari: {df_clean['Hours per day'].mean()}")
print(f"Median dari data berapa jam per hari: {df_clean['Hours per day'].median()}")
print(f"Modus dari data berapa jam per hari: {df_clean['Hours per day'].mode()}")
print(f"Skewness dari berapa jam per hari: {df_clean['Hours per day'].skew()}")
print(f"Kurtosis dari berapa jam per hari: {df_clean['Hours per day'].kurtosis()}")

Mean dari data berapa jam per hari: 3.6991883116883115
Median dari data berapa jam per hari: 3.0
Modus dari data berapa jam per hari: 0    2.0
Name: Hours per day, dtype: float64
Skewness dari berapa jam per hari: 2.46510890599459
Kurtosis dari berapa jam per hari: 10.165535641612383


**Insight**
Dari hasil yang didapat kita mendapatkan bahwa rata-rata responden mendengarkan musik sebanyak 3,69 jam per hari. Namun data memiliki skewness 2.46, diatas 1 yang berarti data ini positively skewed dengan dan highly skewed. Untuk kurtosis hasil menunjukkan bahwa data leptokurtic dan memiliki outlier lebih banyak dibanding normal distribution. Untuk itu, kita mengambil nilai median yang lebih merepresentasikan central tendency dari data ini. Sehingga bisa disimpulkan bahwa median responden mendengarkan lagu sekitar 3 jam per hari.

### **6. Apakah semakin tua umur akan mempengaruhi lamanya mendengarkan lagu?**

Untuk pertanyaan ini akan dijawab dengan menggunakan analisa korelasi antara data umur responden dengan data berapa jam per hari mendengarkan lagu.

In [ ]:
#Mencari Central Tendency pada data umur
print(f"Mean dari data umur: {df_clean['Age'].mean()}")
print(f"Median dari data umur: {df_clean['Age'].median()}")
print(f"Modus dari data umur: {df_clean['Age'].mode()}")
print(f"Skewness dari data umur: {df_clean['Age'].skew()}")
print(f"Kurtosis dari data umur: {df_clean['Age'].kurtosis()}")

Mean dari data umur: 24.800324675324674
Median dari data umur: 21.0
Modus dari data umur: 0    18.0
Name: Age, dtype: float64
Skewness dari data umur: 2.2365613428314806
Kurtosis dari data umur: 5.5298853166996835


Dari hasil yang kita dapat dari normality test skewness dan kurtosis, hasil skewness berada di 2,236 yang lebih besar dari 1. Ini menunjukkan bahwa data umur positively skewed dan tingkat skewnessnya tinggi. Untuk hasil kurtosis berada di 5.52 diatas angka 0, dimana data umur adalah leptokurtic yang mengindikasikan data memiliki outlier lebih banyak dari normal distribution. Untuk data jam sesuai dengan test sebelumnya, tidak terdistribusi dengan normal. maka analisa akan menggunakan Spearman yang lebih robust terhadap outlier.

In [ ]:
corr_rho, pval_s = stats.spearmanr(df_clean['Age'], df_clean['Hours per day']) #memanggil fungsi spearman

print(f"rho-correlation: {corr_rho:.2f}, p-value: {pval_s}")

rho-correlation: -0.14, p-value: 0.0004144545136679447


**Insight**

Dari hasil uji korelasi dengan teknik spearman, didapatkan rho-correlation adalah -0.14 yang mendekati 0 daripada 1, dan p-value yang dibawah critical value 0.05. Hasil rho-correlation yang dibawah 0 ini menunjukkan bahwa terdapat hubungan inverse montonic yang berarti jika umur semakin bertambah, maka jumlah jam yang dihabiskan untuk mendengarkan musik akan berkurang. Sedangkan hasil p-value yang dibawah 0.05 menunjukkan bahwa hubungan ini kebetulan. Namun hasil rho-correlation ini lebih mendekati angka 0 dibanding 1, sehingga dapat disimpulkan juga bahwa hubungan ini lemah.

### **7. Apakah lama waktu mendengarkan musik berpengaruh terhadap efektifitas dalam meringankan gangguan kesehatan mental?**

Kita bisa melihat apakah ada perbedaan berapa jam per hari untuk mendengarkan musik dengan efek dari mendengarkan musik dengan menggunakan ANOVA test.

In [163]:

df_g1 = df_clean[df_clean['Music effects'] == 'Improve']['Hours per day'] #grup 1 berisi data berapa jam per hari untuk responden yang merasa improved
df_g2 = df_clean[df_clean['Music effects'] == 'No effect']['Hours per day']#grup 2 berisi data berapa jam per hari untuk responden yang tidak merasa efek sama sekali
df_g3 = df_clean[df_clean['Music effects'] == 'Worsen']['Hours per day']#grup 3 berisi data berapa jam per hari untuk responden yang merasa memburuk

Untuk menggunakan ANOVA test, antar grup data harus memiliki varian yang homogen. Untuk itu akan digunakan levene test untuk memastikan apakah variance dari masing masing grup kurang lebih sama.

**H0**: Variance antar grup homogen.

**H1**: Variance antar grup tidak homogen.

Test menggunakan confidence level 95% dengan critical value 0.05 (50%) 

In [164]:
#memastikan bahwa variance dari masing masing grup 'efek musik' kurang lebih sama
#test menggunakan levene test
levene_stat, p_val = stats.levene(df_g1, df_g2, df_g3)

print(f'statistic: {levene_stat}')
print(f'p-value: {p_val}')

statistic: 0.10821219772420426
p-value: 0.8974542815968785


p-value > critical value = 0.05, sehingga **H0 fail to reject**. Ini mengindikasikan bahwa variance antar grup homogen. 

ANOVA mengharuskan asumsi grup-grup data memiliki normal distribution, maka dilakukan normality test untuk melihat apakah data masing masing grup terdistribusi normal.

In [165]:
#Tes Skewness dan kurtosis dari masing masing grup
print(f"Skewness dari grup improve: {df_g1.skew()}")
print(f"Kurtosis dari grup improve: {df_g1.kurtosis()}")
print(f"Skewness dari grup no effect: {df_g2.skew()}")
print(f"Kurtosis dari grup no effect: {df_g2.kurtosis()}")
print(f"Skewness dari grup worsen: {df_g3.skew()}")
print(f"Kurtosis dari grup worsen: {df_g3.kurtosis()}")

Skewness dari grup improve: 2.397264214458892
Kurtosis dari grup improve: 10.081666736980974
Skewness dari grup no effect: 2.7269803838297406
Kurtosis dari grup no effect: 11.093802424584378
Skewness dari grup worsen: 2.153296748795532
Kurtosis dari grup worsen: 5.545596655086219


Disini terlihat bahwa skewness setiap grup berada diatas 1, yang menandakan bahwa setiap grup positively skewed dan memiliki tingkat skewness yang tinggi. Kurtosis setiap grup juga diatas 0, menandakan leptokurtic. Untuk itu test tidak bisa menggunakan anova dikarenakan tidak bisa memenuhi asumsi normality. Maka akan digunakan test kruskal-wallis test sebagai pengganti ANOVA test.

Melakukan test kruskal-wallis

**H0**: Tidak ada perbedaan median antar grup.

**H1**: Ada perbedaan median antar grup.

Test menggunakan confidence level 95% dengan critical value 0.05 (50%) 

In [166]:
print(f'Median dari jumlah jam per hari yang berefek Improve: {df_g1.mean()}')
print(f'Median dari jumlah jam per hari yang berefek No effect: {df_g2.mean()}')
print(f'Median dari jumlah jam per hari yang berefek Worsen: {df_g3.mean()}')

Median dari jumlah jam per hari yang berefek Improve: 3.763841201716738
Median dari jumlah jam per hari yang berefek No effect: 3.561111111111111
Median dari jumlah jam per hari yang berefek Worsen: 2.933333333333333


In [167]:
kruska_stat, p_val  = stats.kruskal(df_g1,df_g2,df_g3) #memanggil fungsi kruskal-wallis test
print(f"P-value: {p_val}")
print(f"Statistic: {kruska_stat}")

P-value: 0.07960251332987078
Statistic: 5.0614192242688


**Insight**

Dari hasil p-value dan statistic yang didapatkan, p-value berada diatas critical value 0.05, sehingga H0 failed to reject. Ini mengindikasikan bahwa tidak ada perbedaan median berapa jam per hari mendengar musik pada setiap kategori hasil efek mendengarkan musik. Asumsi yang bisa didapat adalah lama waktu mendengarkan tidak bisa dianggap menjadi variable utama yang berpengaruh terhadap efek mendengarkan musik, namun asumsi ini perlu dibuktikan lebih dalam. 

### **8. Seperti Apa Persebaran data BPM musik terhadap masing masing efek dari mendengarkan musik?**

In [169]:
df_g1 = df_clean[df_clean['Music effects'] == 'Improve']['BPM'] #grup 2 berisi data BPM untuk responden yang merasa improved
df_g2 = df_clean[df_clean['Music effects'] == 'No effect']['BPM'] #grup 1 berisi data BPM untuk responden yang merasa improved
df_g3 = df_clean[df_clean['Music effects'] == 'Worsen']['BPM'] #grup 1 berisi data BPM untuk responden yang merasa improved

In [192]:
fig = px.histogram(df_g1, x='BPM',title= 'Data distribution for BPM data for Improved effect')
fig.show()
print(f'Skewness : {df_g1.skew()}')
print(f"Kurtosis: {df_g1.kurtosis()}")

Skewness : 3.684509158727729
Kurtosis: 44.11346518937343


Data BPM untuk hasil 'improved' terlihat positively skewed dengan satu data memiliki BPM value lebih dari 600. Data banyak terdistribusi di range 80 BPM sampai 170 BPM.

In [193]:
fig = px.histogram(df_g2, x='BPM', title='Data distribution for BPM data for No effect')
fig.show()
print(f'Skewness : {df_g2.skew()}')
print(f"Kurtosis: {df_g2.kurtosis()}")

Skewness : 0.05451759983129552
Kurtosis: 1.2334931045895021


Data BPM untuk hasil 'No effect' terlihat hampir normally distributed dengan kurtosis yang leptokurtic dengan data cukup tinggi terkumpul di range 120 BPM ke 129 BPM

In [194]:

fig = px.histogram(df_g3, x='BPM', title='Data distribution for BPM data for Worsen effect')
fig.show()
print(f'Skewness : {df_g3.skew()}')
print(f"Kurtosis: {df_g3.kurtosis()}")

Skewness : 0.03146782159975531
Kurtosis: 0.527993643641012


Data BPM untuk hasil 'Worsen' terlihat normally distributed dengan berkumpul data di range 100 BPM - 139 BPM

In [199]:
fig = px.histogram(df_clean, x='BPM', color='Music effects', hover_data=df_clean.columns, title='Data distribution for BPM data')
fig.show()
print(f'Skewness : {df_clean["BPM"].skew()}')
print(f"Kurtosis: {df_clean['BPM'].kurtosis()}")

Skewness : 3.247953991720796
Kurtosis: 40.93352955234013


**Insight**

Secara visual, terlihat bahwa data masing masing grup hasil efek musik berkumpul di range 80 BPM - 170 BPM.

## **6. Pengambilan Kesimpulan**

Dari insight yang didapatkan melalui analisa statistik dan explorasi data, terdapat beberapa point yang bisa kita dapat:
* 75,6% Responden merasa ada peningkatan pada penyembuhan penyakit mental mereka, diatas dari target 50% total pasien pada peningkatan penyembuhan.

* Lagu Rock, Pop, dan Metal menjadi genre yang disenangi sebagian besar responden, dengan Lofi, gospel, dan latin menjadi yang terendah.
* Genre lagu tidak memiliki hubungan dengan efek yang didapat dalam mendengarkan musik.
* Median responden mendengarkan lagu adalah sekitar 3 jam per hari
* Semakin bertambah umur, maka semakin singkat waktu yang dipakai untuk mendengarkan lagu.
* Adanya asumsi bahwa lama waktu mendengarkan musik tidak bisa menjadi acuan efektifitas musik dalam penyembuhan penyakit mental.
* Responden banyak yang mendengarkan lagu 80 BPM - 170 BPM, BPM yang ada untuk lagu yang bersifat santai hingga ke lagu yang memiliki tempo cukup cepat seperti lagu metal, rock, maupun techno.

Dari point-point ini bisa dibuatkan kesimpulan bahwa penggunaan musik sebagai suplemen dalam terapi penyembuhan penyakit mental dapat dilakukan dikarenakan sebagian besar responden dapat merasakan efek dari aktifitas tersebut. Namun penggunaan musik tidak bisa dijadikan prosedur medis dikarenakan kurangnya bukti statistik dari dataset yang didapat yang mampu menyokong laporan efektifitas responden itu sendiri. Ada kemungkinan bahwa responden mengalami 'placebo effect' atau dibutuhkan pengumpulan data secara lebih detil dan analisa lebih dalam mengenai hubungan musik dengan kesehatan mental ini.

Namun hasil ini tidak menutup kemungkinan bahwa mendengarkan musik sebagai suatu aktifitas yang mengapresiasikan seni bisa memberikan efek positif terhadap keseimbangan jiwa, sehingga tidak ada masalah untuk merekomendasikannya kepada pasien dengan jenis lagu dan lama waktu mendengarkan yang dapat disesuaikan sesuai keinginan dan kebiasaan pasien.